dtfs-central-api/src/v1/validation/utilisation-report-service/fee-record-validator.test.ts

```diff
@@ -0,0 +1,174 @@
+      return FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withPaymentCurrency(paymentCurrency).build();
+    }
+  });
+
+  describe('validateFeeRecordsFormCompleteGroup', () => {
+    const feeRecordFindBySpy = jest.spyOn(FeeRecordRepo, 'findByIdWithPaymentsAndFeeRecords');
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    describe(`when all fee records in the group have status ${FEE_RECORD_STATUS.TO_DO}`, () => {
+      it('should not throw', async () => {
+        // Arrange
+        const payments = [
+          PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+            .withFeeRecords([
+              FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(1).build(),
+              FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(2).build(),
+            ])
+            .build(),
+        ];
+        const feeRecords = [
+          FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(1).withStatus(FEE_RECORD_STATUS.TO_DO).withPayments(payments).build(),
+          FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(2).withStatus(FEE_RECORD_STATUS.TO_DO).withPayments(payments).build(),
+        ];
+        feeRecordFindBySpy.mockResolvedValue(feeRecords);
+
+        const feeRecordIds = [1, 2];
+
+        // Act + Assert
+        await expect(validateFeeRecordsFormCompleteGroup(feeRecordIds)).resolves.not.toThrow();
+      });
+    });
+
+    describe.each(difference(Object.values(FEE_RECORD_STATUS), [FEE_RECORD_STATUS.TO_DO]))('when all fee records in the group have status %s', (status) => {
+      describe('and the fee record ids match those of the fee record payment group', () => {
+        it('should not throw', async () => {
+          // Arrange
+          const payments = [
+            PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP)
+              .withFeeRecords([
+                FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(1).build(),
+                FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(2).build(),
+              ])
+              .build(),
+          ];
+          const feeRecords = [
+            FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(1).withStatus(status).withPayments(payments).build(),
+            FeeRecordEntityMockBuilder.forReport(aReconciliationInProgressReport()).withId(2).withStatus(status).withPayments(payments).build(),
+          ];
+          feeRecordFindBySpy.mockResolvedValue(feeRecords);
+
+          const feeRecordIds = [1, 2];
+
+          // Act + Assert
+          await expect(validateFeeRecordsFormCompleteGroup(feeRecordIds)).resolves.not.toThrow();
+        });
+      });
+
+      describe('and there are missing fee record ids from the fee record payment group', () => {
```

Comment:
What about `and there are additional fee record ids than those in the payment group`?